modules/nf-core/mageck/count/main.nf

```diff
@@ -0,0 +1,43 @@
+process MAGECK_COUNT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::mageck=0.5.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mageck:0.5.9--py37h6bb024c_0':
+        'quay.io/biocontainers/mageck:0.5.9--py37h6bb024c_0' }"
+
+    input:
+    tuple val(meta), path(inputfile)
+    path(library)
+
+    output:
+    tuple val(meta), path("*count*.txt"), emit: count
+    tuple val(meta), path("*.count_normalized.txt"), emit: norm
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def input_file = ("$inputfile".endsWith(".fastq.gz")) ? "--fastq ${inputfile}" : "-k ${inputfile}"
+    def sample_label = ("$inputfile".endsWith(".fastq.gz")) ? "--sample-label ${meta.id}" : ''
+
+    """
+    mageck \\
+        count \\
+        $args \\
+        -l $library \\
+        -n $prefix \\
+        $sample_label \\
+        $input_file \\
```

Comment:
A quick check of the `mageck count` docs shows that you may need multiple input files `e.g. condition and control`, is it always exactly 2 files or can this number vary? How will you provide those to the module?

Reply:
Mageck count takes as many fastq files and conditions, eg `--sample-label A_ctrl,B_ctrl,C_ctrl` and then the corresponding fastq file (usually there's no paired end in these kind of sequencing), so i was planning to collect the fastq files and the meta info from the sample sheet and provide them to the module in my pipeline. Would this work with how I've implemented the module?

Reply:
ok I see, also in the test it's done like that. Yes it might be possible to get the channels like this with some previous channel magic, and otherwise you can always update the module if needed! 😄